elastic/kibana
# [Mappings editor] Re-enable scaled float test

This PR unskips the `scaled_float_datatype` tests. Changes include minor improvements to the tests, as well as removal of the deprecated `FormDataProvider` in favor of the `useFormData` hook.

Note: It's not possible to run jest tests via the flaky test runner, but I've run this through CI several times.

Fixes https://github.com/elastic/kibana/issues/145102

### How to test
There should be no functional changes; the `numeric` field type should work as before.

1. Navigate to Index Management > Index Templates
2. Create a new index template
3. Verify you are able to add a Numeric > Scaled float field type via the "Mappings" step.

<img width="1283" alt="Screenshot 2023-08-22 at 11 37 58 AM" src="https://github.com/elastic/kibana/assets/5226211/070f04e1-cd3a-4499-8e5e-cb349276dd8b">
